tests/csapi/apidoc_room_create_test.go

```diff
@@ -61,6 +62,78 @@ func TestRoomCreate(t *testing.T) {
 			})
 			content := alice.MustGetStateEventContent(t, roomID, "m.room.topic", "")
 			must.MatchGJSON(t, content, match.JSONKeyEqual("topic", "Test Room"))
+
+			// Rich topics not implemented yet on Dendrite
+			runtime.SkipIf(t, runtime.Dendrite)
+
+			// The plain text topic is duplicated into m.topic
+			must.MatchGJSON(t, content,
+				match.JSONKeyArrayOfSize("m\\.topic.m\\.text", 1),
+				match.JSONKeyPresent("m\\.topic.m\\.text.0.body"),
+				match.JSONKeyEqual("m\\.topic.m\\.text.0.body", "Test Room"))
+
+			// The mime type must be unset or text/plain
+			mime := content.Get("m\\.topic.m\\.text.0.mimetype")
+			if mime.Exists() {
+				must.Equal(t, mime.String(), "text/plain", "no m.topic content block")
+			}
+		})
+		// POST /createRoom makes a room with a topic via initial_state
+		t.Run("POST /createRoom makes a room with a topic via initial_state", func(t *testing.T) {
+			t.Parallel()
+
+			roomID := alice.MustCreateRoom(t, map[string]interface{}{
+				"initial_state": []map[string]interface{}{
+					{
+						"content": map[string]interface{}{
+							"topic": "Test Room",
+						},
+						"type":      "m.room.topic",
+						"state_key": "",
+					},
+				},
+				"preset": "public_chat",
+			})
+			content := alice.MustGetStateEventContent(t, roomID, "m.room.topic", "")
+			must.MatchGJSON(t, content, match.JSONKeyEqual("topic", "Test Room"))
+
+			// There is no m.topic property
+			must.MatchGJSON(t, content, match.JSONKeyMissing("m\\.topic"))
+		})
+		// POST /createRoom makes a room with a topic via initial_state overwritten by topic
+		t.Run("POST /createRoom makes a room with a topic via initial_state overwritten by topic", func(t *testing.T) {
+			t.Parallel()
+
+			roomID := alice.MustCreateRoom(t, map[string]interface{}{
+				"topic": "Test Room",
+				"initial_state": []map[string]interface{}{
+					{
+						"content": map[string]interface{}{
+							"topic": "Shenanigans",
+						},
+						"type":      "m.room.topic",
+						"state_key": "",
+					},
+				},
+				"preset": "public_chat",
+			})
+			content := alice.MustGetStateEventContent(t, roomID, "m.room.topic", "")
+			must.MatchGJSON(t, content, match.JSONKeyEqual("topic", "Test Room"))
+
+			// Rich topics not implemented yet on Dendrite
+			runtime.SkipIf(t, runtime.Dendrite)
```

Comment:
We should be skipping at the top of this test

Reply:
Sorry, I had assumed `SkipIf` to only skip the remainder of the test but it looks like it marks the whole test as skipped unless it fails. Have move the statement to the top with e61bcb72e52cec903aad34e102d16d37d1549069.